warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java

```diff
@@ -1177,22 +1178,33 @@ protected ASTJexlScript processTree(final ASTJexlScript originalQueryTree, Shard
             
             // Expand any bounded ranges into a conjunction of discrete terms
             try {
-                innerStopwatch = timers.newStartedStopwatch("DefaultQueryPlanner - Expand regex");
+                // Count each type of node in the query.
+                NodeTypeCountVisitor nodeCount = NodeTypeCountVisitor.countNodes(queryTree);
+                
                 ParallelIndexExpansion regexExpansion = new ParallelIndexExpansion(config, scannerFactory, metadataHelper, expansionFields,
                                 config.isExpandFields(), config.isExpandValues(), config.isExpandUnfieldedNegations());
-                queryTree = (ASTJexlScript) regexExpansion.visit(queryTree, null);
-                if (log.isDebugEnabled()) {
-                    logQuery(queryTree, "Query after expanding regex:");
+                if (nodeCount.hasRegexNodes()) {
+                    innerStopwatch = timers.newStartedStopwatch("DefaultQueryPlanner - Expand regex");
+                    queryTree = (ASTJexlScript) regexExpansion.visit(queryTree, null);
+                    if (log.isDebugEnabled()) {
+                        logQuery(queryTree, "Query after expanding regex:");
+                    }
+                    innerStopwatch.stop();
+                } else {
+                    log.debug("No regex to expand");
                 }
-                innerStopwatch.stop();
                 
-                innerStopwatch = timers.newStartedStopwatch("DefaultQueryPlanner - Expand ranges");
-                queryTree = RangeConjunctionRebuildingVisitor.expandRanges(config, scannerFactory, metadataHelper, queryTree, config.isExpandFields(),
-                                config.isExpandValues());
-                if (log.isDebugEnabled()) {
-                    logQuery(queryTree, "Query after expanding ranges:");
+                if (nodeCount.hasPossibleBoundedRange()) {
```

Comment:
Given #926 has been merged in, we can now detect bounded ranges precisely.  Update the node counter to also count the different types of QueryPropertyMarkers.  This can then be if (nodeCount.asBoundedRange()).

Reply:
Done.